packages/payment-processor/src/payment/utils-near.ts

```diff
@@ -77,3 +80,57 @@ export const processNearPayment = async (
     throw new Error(`Could not pay Near request. Got ${e.message}`);
   }
 };
+
+/**
+ * Export used for mocking only.
+ */
+export const processNearPaymentWithConversion = async (
```

Comment:
I think this one deserves a bit of documentation for amount decimals (fiat at least), as well as the timespan unit, the currency... Similar to the contract itself.